state/secrets.go

```diff
@@ -905,6 +965,139 @@ func (st *State) WatchConsumedSecretsChanges(consumer string) StringsWatcher {
 	})
 }
 
+// WatchObsoleteRevisions returns a watcher for notifying
+// when secret revisions no longer have any consumers wanting
+// to use them.
+func (st *State) WatchObsoleteRevisions(owner string) StringsWatcher {
+	return newCollectionWatcher(st, colWCfg{
+		col: secretRevisionsC,
+		filter: func(key interface{}) bool {
+			secretMetadataCollection, closer := st.db().GetCollection(secretMetadataC)
+			defer closer()
+
+			secretRevisionsCollection, closer := st.db().GetCollection(secretRevisionsC)
+			defer closer()
+
+			var doc secretRevisionDoc
+			err := secretRevisionsCollection.FindId(key).One(&doc)
+			if err != nil {
+				return false
+			}
+			parts := strings.Split(st.localID(doc.DocID), "/")
+
+			var mdDoc secretMetadataDoc
+			err = secretMetadataCollection.FindId(parts[0]).Select(bson.D{{"owner-tag", 1}}).One(&mdDoc)
+			if err != nil {
+				return false
```

Comment:
here too

Reply:
yeah, it's how our watchers are designed